sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/InitializeObjectsWithValueBase.cs

```diff
@@ -119,19 +122,21 @@ private bool IsLaterAssignedWithExpectedValue(ObjectCreationExpressionSyntax obj
             bool TrackedPropertySetWithExpectedValue(AssignmentExpressionSyntax assignment) =>
                 variableSymbol.Equals(GetAssignedVariableSymbol(assignment, semanticModel))
                 && IsTrackedPropertyName(assignment.Left)
-                && ValueIsExpected(assignment, semanticModel);
+                && IsExpectedValue(assignment.Right, semanticModel);
         }
 
         private bool IsTrackedType(ExpressionSyntax expression, SemanticModel semanticModel) =>
             semanticModel.GetSymbolInfo(expression).Symbol
                 .GetSymbolType()
                 .Is(TrackedType);
 
-        private bool IsInitializedAsExpected(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
+        protected virtual bool IsInitializedAsExpected(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
```

Comment:
The ctor+initializer check is different, hence the virtual - previously we assumed that the default ctor has the wrong value, but in this rule it is with the correct value